status-im/status-react
# Fix JS lock files, switch to yarn

Fixes #2541

### Summary:

There was a lot of funny interactions between `npm install`, and the `package-lock.json` file. `npm install` was modifying the lock file, and the `postinstall.sh` script was doing a bunch of extra weird stuff external to the lock file that it shouldn't be doing. As well, subsequent runs of `npm install` were not intelligently checking whether `node_modules` was up to date, and was doing a full rebuild, slowing down `scripts/setup`.

I am currently waiting for #2574 to land, so I can merge the setup script updates that I made in here with those changes, so this is pending that.

### Steps to test:

1. Check out this branch
2. Run `scripts/setup`
3. It should `yarn install` successfully

status: wip
